packages/schematics/src/command-line/affected-apps.ts

```diff
@@ -60,7 +60,9 @@ function directlyTouchedProjects(
   return projects
     .filter(project => {
       return touchedFiles.some(file => {
-        return file.startsWith(project.root);
+        return project.files.some(projectFile => {
+          return file.endsWith(projectFile);
```

Comment:
Correct me if I'm wrong but in the following case won't all projects be touched?
```ts
file = 'tsconfig.json'

project1.files = [
  'apps/app1/tsconfig.json'
];
project2.files = [
  'apps/app2/tsconfig.json'
];
```